tarantool_test.go

```diff
@@ -691,6 +692,59 @@ func TestClient(t *testing.T) {
 	if val != 11 {
 		t.Errorf("5 + 6 == 11, but got %v", val)
 	}
+
+	// Check for skip SQL tests if tarantool version < 2.0.0
+	resp, err = conn.Eval("return box.info.version", []interface{}{})
+	if err != nil {
+		t.Errorf("Failed to Eval for checking Tarantool version")
+	}
+	ver := parseTarantoolVersion(resp.Data[0].(string))
+	if !sqlSupported(ver) {
+		return
+	}
+
+	// SQL Execute
+	// prepare map for sql bind
+	sqlBind := map[string]interface{}{
+		"id":   1,
+		"name": "test",
+	}
+	// insert data using sql query
+	resp, err = conn.Execute("INSERT INTO SQL_TEST VALUES (:id, :name);", sqlBind)
```

Comment:
To be honest, I'm not well-versed in Tarantool SQL, but I think it's better to test it at least will all basic operations (now we lack `CREATE`, `DELETE`, `DROP`):
https://www.tarantool.io/ru/doc/latest/tutorials/sql_tutorial/

Reply:
I like the idea about covering all basic cases. I would add here a bit.

Since we aims to test implementation of the protocol rather than SQL dialect implementation, I propose to cover the basic cases of the protocol.

* Generally: successful response and one that states a failure (e.g. duplicate key).
* A response that returns a result set (an array of tuples). SELECT is the most important example.
  * With and without bindings. With named and numerical bindings.
  * With diffferent metadata:
    * Different column types.
    * With/without nullability / collation.
    * With/without autoincremented rows.
    * With/without extended metainfo.
      * When the extended metainfo is enabled: span == name and span != name.
    * When a column is directly from a table (`select X from T`) and not (`select X*2 from T`).
* A response that does not return a result set, just some meta. INSERT and UPDATE are the most important here.
  * With zero modified rows / with some modified rows. Consider also the case, when a row is modified to the same value as it has before.
  * With/without autogenerated primary key. AFAIR, those keys are always integers and never compound keys, but, please, re-check it.
  * Some DDL (CREATE/ALTER/DROP TABLE): the response format is the same as for INSERT, but it worth to freeze what is reported as modified rows count (and autogenerated keys as well).

Take care to the API for obtaining the response information:

* Convenience for a Go developer.
* Direct mapping from the protocol.

Aside of the protocol documentation you're surely aware, I would also highlight the [old RFC document](https://github.com/tarantool/tarantool/blob/3b028af7ab08d66749d77749b276f8148af7b000/doc/rfc/3328-wire_protocol.md). The final implementation differs from it in exact names and details (such as autoincremented IDs), but it is good to understand thought behind the protocol. You can also be interested in motivation behind the 'span' in the metadata, see https://github.com/tarantool/tarantool/issues/4407. There also were some changes around how columns are named if they're not originated directly from a table column: https://github.com/tarantool/tarantool/issues/3962.

Please, keep the test cases well structured (a reader understands where one ends and another one starts) and well commented (a reader understands what is the predicate, which is hold by given test case).

Reply:
Thank you for ideas and references! Added tests for enumerated cases.